### DISCUSSION TITLE
Eigen matrix as a default parameter

### DISCUSSION THREAD

**Discussion by contagon** [wjakob/nanobind#166] [Q&A] [answered]
In my attempts to port from pybind11 to nanobind I'm hitting a bit of a kink. I'm attempting to add an Eigen matrix as a default parameter to a constructor as something like this:
```cpp
.def(nb::init<Eigen::Matrix2d>(), "mat"_a=Eigen::Matrix2d::Identity())
```
which compiles fine, but upon importing the module in python, I get the following error,
```bash
*** stack smashing detected ***: terminated
```
Removing the default parameter removes this issue and everything seems to function properly. I wanted to check if this was supposed to be supported in nanobind? If so, I'd be happy to put together a small working example of this bug. If not, if you wouldn't mind providing a proper way of doing this, that'd be great.

Thanks so much!

**Comment by wjakob** [comment, marked as answer]
Seems suspicious, I think that should work. Please go ahead with the reproducer.

**Reply by wjakob** [reply]
Please double-check that you have NumPY installed in your virtual environment. Otherwise that might be a reason (though it really should not crash like that).

**Reply by contagon** [reply]
K, just gave it another look. It appears to only be an issue with 1x1 matrices, although it's different that the stack smash I was seeing previously (was an unrelated issue). I realize a 1x1 matrix may seem silly, but it does have its uses in my project due to how things are templated.

Here's a minimal example based on the nanobind_example repo ([here's my fork](https://github.com/contagon/nanobind_example)),
```cpp
#include <Eigen/Core>

#include <nanobind/nanobind.h>
#include <nanobind/eigen/dense.h>

namespace nb = nanobind;

using namespace nb::literals;
using namespace Eigen;

Matrix<double,1,1> add(Matrix<double,1,1> a=Matrix<double,1,1>::Zero(), Matrix<double,1,1> b=Matrix<double,1,1>::Zero()){
    return a + b;
}

NB_MODULE(nanobind_example_ext, m) {
    m.def("add", &add, "a"_a=Matrix<double,1,1>::Zero(), "b"_a=Matrix<double,1,1>::Zero());
}
```
After a `pip install .` I get the following error when trying to run
```
$ python -c 'import nanobind_example; print(nanobind_example.add())'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: add(): incompatible function arguments. The following argument types are supported:
    1. add(a: numpy.ndarray[dtype=float64, shape=(1, 1), order='F'] = 0, b: numpy.ndarray[dtype=float64, shape=(1, 1), order='F'] = 0) -> numpy.ndarray[dtype=float64, shape=(1, 1), order='F']

Invoked with types:
```
Is this a small edge case that may be slipping through? It looks like the default argument got set to 0 rather than [ 0 ]. Or if you seem something else obviously wrong here, please let me know.

**Reply by wjakob** [reply]
Fixed in 445781fc2cf2fa326cc22e8fd483e8e4a7bf6cf5

**Reply by contagon** [reply]
Late response, but thanks so much for the quick fix! 100% fixed it for me.
